go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -229,53 +228,23 @@ func (oc *Controller) addEgressFirewall(egressFirewall *egressfirewallapi.Egress
 	}
 
 	ipv4HashedAS, ipv6HashedAS := nsInfo.addressSet.GetASHashNames()
-	err = oc.addEgressFirewallRules(ipv4HashedAS, ipv6HashedAS, egressFirewall.Namespace, egressFirewallStartPriorityInt)
+	err = oc.addEgressFirewallRules(ipv4HashedAS, ipv6HashedAS, egressFirewall.Namespace, egressFirewallStartPriorityInt, txn)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (oc *Controller) updateEgressFirewall(oldEgressFirewall, newEgressFirewall *egressfirewallapi.EgressFirewall) error {
-	// block all external traffic in this namespace
-	nsInfo, err := oc.waitForNamespaceLocked(newEgressFirewall.Namespace)
-	if err != nil {
-		return fmt.Errorf("cannot update egressfirewall in %s:%v", newEgressFirewall.Namespace, err)
-	}
-	addressSet := nsInfo.addressSet
-	nsInfo.Unlock()
-	priority, err := strconv.Atoi(types.EgressFirewallStartPriority)
-	if err != nil {
-		return fmt.Errorf("cannot update egressfirewall in %s:%v", newEgressFirewall.Namespace, err)
-	}
-
-	ipv4ASHashName, ipv6ASHashName := addressSet.GetASHashNames()
-	match := generateMatch(ipv4ASHashName, ipv6ASHashName, []matchTarget{
-		{matchKindV4CIDR, "0.0.0.0/0"},
-		{matchKindV6CIDR, "::/0"},
-	},
-		[]egressfirewallapi.EgressFirewallPort{},
-	)
-
-	err = oc.createEgressFirewallRules(priority, match, "drop", newEgressFirewall.Namespace+"-blockAll")
-	if err != nil {
-		return fmt.Errorf("cannot update egressfirewall in %s:%v", newEgressFirewall.Namespace, err)
-	}
-
-	updateErrors := oc.deleteEgressFirewall(oldEgressFirewall)
+func (oc *Controller) updateEgressFirewall(oldEgressFirewall, newEgressFirewall *egressfirewallapi.EgressFirewall, txn *util.NBTxn) error {
+	updateErrors := oc.deleteEgressFirewall(oldEgressFirewall, txn)
 	// add the new egressfirewall
 
-	updateErrors = errors.Wrapf(updateErrors, "%v", oc.addEgressFirewall(newEgressFirewall))
-	// delete rules blocking all external traffic
-	err = oc.deleteEgressFirewallRules(newEgressFirewall.Namespace + "-blockAll")
```

Comment:
@JacobTanenbaum looks like -blockAll doesn't get used anymore? Is that intended?

Reply:
Aha, that's the "don't need to block all traffic during upgrades" thing from the PR description right?

Reply:
yes